lib/Fmt_ast.ml

```diff
@@ -2914,26 +2906,46 @@ and fmt_expression c ?(box = true) ?(pro = noop) ?eol ?parens
       pro $ fmt_indexop_access c ctx ~fmt_atrs ~has_attr ~parens x
   | Pexp_hole -> pro $ hvbox 0 (fmt_hole () $ fmt_atrs)
   | Pexp_beginend e ->
-      fmt_beginend c ~box ~pro ~ctx ~fmt_atrs ~ext ~indent_wrap ?eol e
+      fmt_beginend c ~loc:pexp_loc ~box ~pro ~ctx ~fmt_atrs ~ext ~indent_wrap
+        ?eol e
   | Pexp_parens e ->
       pro
       $ hvbox 0
           (fmt_expression c ~box ?eol ~parens:true ~indent_wrap ?ext
              (sub_exp ~ctx e) )
       $ fmt_atrs
 
-and fmt_beginend c ?(box = true) ?(pro = noop) ~ctx ~fmt_atrs ~ext
+and fmt_lazy c ~ctx ?(pro = noop) ~fmt_atrs ~ext ~parens e =
+  let lazy_ = str "lazy" $ fmt_extension_suffix c ext in
+  match e with
+  | {pexp_desc= Pexp_beginend _; _} as e ->
```

Comment:
Duplicating the code this way is not sustainable. The difference between the two cases is the usual `inner, outer` thing so the code can easily be written with more sharing.

Reply:
This is done.